src/world/Spell/Spell.h

```diff
@@ -1813,10 +1815,39 @@ class SpellTargetConstraint
                 GameobjectTargets.push_back(id);
         }
 
+		void AddFocused(uint32 value,int type)
+		{
+			m_TargetFocus.insert(std::pair< uint32, uint32 >(value, type));
+		}
+
+		bool IsFocused(uint32 value)
+		{
+			TargetFocusMap::const_iterator itr = m_TargetFocus.find(value);
+
+			if (itr != m_TargetFocus.end())
+				return itr->second;
+			else
+				return nullptr;
```

Comment:
You can not return a pointer (in this case nullptr) if the function returns a bool.